tests/testsuite.src/syn_misc.at

```diff
@@ -8414,3 +8414,60 @@ prog.cob:18: error: ANY LENGTH items may only be BY REFERENCE formal parameters
 
 AT_CLEANUP
 
+
+AT_SETUP([IMP INCLUDE directive])
+AT_KEYWORDS([IMP INCLUDE])
+
+AT_DATA([prog.cob], [
+       >>IMP INCLUDE "file.h"
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. prog.
+       PROCEDURE DIVISION.
+           GOBACK.
+])
+
+AT_CHECK([$COMPILE_ONLY prog.cob], [0], [], [])
+
+AT_DATA([prog.cob], [
+       >>IMP INCLUDE <file.h>
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. prog.
+       PROCEDURE DIVISION.
+           GOBACK.
+])
+
+AT_CHECK([$COMPILE_ONLY prog.cob], [0], [], [])
+
+AT_DATA([prog2.cob], [
+       >>IMP INCLUDE
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. prog.
+       PROCEDURE DIVISION.
+           GOBACK.
+])
+
+AT_CHECK([$COMPILE_ONLY prog2.cob], [1], [],
+[prog2.cob:2: error: invalid IMP INCLUDE directive
+])
+
+AT_CLEANUP
+
+
+AT_SETUP([IMP INCLUDE directive multiple files])
+AT_KEYWORDS([IMP INCLUDE])
+
+AT_DATA([prog.cob], [
+       >>IMP INCLUDE "file1.h" file2.h <file3.h> "<file4.h>"
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. PROG.
+       PROCEDURE DIVISION.
+           GOBACK.
+])
+
```

Comment:
If you add prog2.cob here without any 2>>IMP INCLUDE" the prog2.c has those includes as well (or cobc aborts), no?

I _think_ `cb_include_file_list_directive` should be reset to `NULL` before the next non-nested program is generated, similar to how the source format is reset.

Reply:
I'm not sure to understand. Are you thinking of something like this:

```cobol
       >>IMP INCLUDE "file1.h" file2.h <file3.h> "<file4.h>"
       IDENTIFICATION DIVISION.
       PROGRAM-ID. prog1.
       PROCEDURE DIVISION.
           STOP RUN.
       END PROGRAM prog1.

       IDENTIFICATION DIVISION.
       PROGRAM-ID. prog2.
       PROCEDURE DIVISION.
           STOP RUN.
       END PROGRAM prog2.
```
?

In this case, the includes only occur once.

Reply:
No, something like `prog1.cob prog2.cob`, which will generate two independent C files.